presto-main/src/main/java/com/facebook/presto/cost/ComposableStatsCalculator.java

```diff
@@ -66,21 +66,26 @@ public ComposableStatsCalculator(List<Rule> rules)
     @Override
     public PlanNodeStatsEstimate calculateStats(PlanNode node, StatsProvider sourceStats, Lookup lookup, Session session, Map<Symbol, Type> types)
     {
-        Iterator<Rule> ruleIterator = getCandidates(node).iterator();
+        Iterator<Rule<?>> ruleIterator = getCandidates(node).iterator();
         while (ruleIterator.hasNext()) {
-            Rule rule = ruleIterator.next();
-            Optional<PlanNodeStatsEstimate> calculatedStats = rule.calculate(node, sourceStats, lookup, session, types);
+            Rule<?> rule = ruleIterator.next();
+            Optional<PlanNodeStatsEstimate> calculatedStats = calculatedStats(rule, node, sourceStats, lookup, session, types);
             if (calculatedStats.isPresent()) {
                 return calculatedStats.get();
             }
         }
         return PlanNodeStatsEstimate.UNKNOWN_STATS;
     }
 
-    public interface Rule
+    private static <T extends PlanNode> Optional<PlanNodeStatsEstimate> calculatedStats(Rule<T> rule, PlanNode node, StatsProvider sourceStats, Lookup lookup, Session session, Map<Symbol, Type> types)
```

Comment:
I would call the method with a verb, since this invokes action (`calculateStats` rather than `calculatedStats`)